api/server/sdk/server_interceptors.go

```diff
@@ -54,8 +54,22 @@ func (s *sdkGrpcServer) rwlockIntercepter(
 
 // Authenticate user and add authorization information back in the context
 func (s *sdkGrpcServer) auth(ctx context.Context) (context.Context, error) {
+	var token string
+	var err error
+
+	// public call attempted, add system.public user
+	if auth.IsPublic(ctx) {
+		return auth.ContextSaveUserInfo(ctx, &auth.UserInfo{
```

Comment:
Create a function in auth to return the UserInfo for public with the appropriate RBAC.

Pass in the value of the PublicVolumeCreationDisabled bool into the new function and let it set the rules for the system.public role accordingly.

Reply:
We decided to check PublicVolumeCreationDisabled on server startup and updating the system.public user RBAC accordingly.